common/test/acceptance/pages/studio/settings.py

```diff
@@ -61,8 +63,13 @@ def save_changes(self, wait_for_confirmation=True):
                 'Waiting for save confirmation...'
             ).fulfill()
 
-    def refresh_page(self):
+    def refresh_page(self, wait_for_confirmation=True):
         """
         Reload the page.
         """
         self.browser.refresh()
+        if wait_for_confirmation:
+            EmptyPromise(
+                lambda: self.q(css='body.view-settings').present,
```

Comment:
Does this really signify that the page is there, and all the javascript has been loaded, and any scripts executed? 
I could be wrong, but I don't think this is the case. I just looked now, and maybe you got this from the is_browser_on_page method? That method itself could be misleadingly flaky. 

Perhaps the lambda is self.is_browser_on_page and also fix the is_browser_on_page method.